src/dex/jarvis-v6/utils.ts

```diff
@@ -0,0 +1,123 @@
+import { ethers } from 'ethers';
+import { IDexHelper } from '../../dex-helper';
+import { Address, Token } from '../../types';
+import { bigIntify } from '../nerve/utils';
+import { getBigIntPow } from '../../utils';
+import { JarvisSwapFunctions, PoolConfig, PoolState } from './types';
+import SynthereumPriceFeedABI from '../../abi/jarvis/SynthereumPriceFeed.json';
+import { Interface } from '@ethersproject/abi';
+import { BI_POWS } from '../../bigint-constants';
+
+export const THIRTY_MINUTES = 60 * 30;
+
+export async function getOnChainState(
+  dexHelper: IDexHelper,
+  priceFeedAddress: Address,
+  poolConfigs: PoolConfig[],
+  poolInterface: Interface,
+  blockNumber: number | 'latest',
+): Promise<PoolState[]> {
+  const pricesFeedResults = await _getPricesFeedValue(
+    dexHelper,
+    poolConfigs,
+    priceFeedAddress,
+    blockNumber,
+  );
+
+  const multiContract = dexHelper.multiContract;
+  const PoolCallData = poolConfigs
+    .map(pool => [
+      {
+        target: pool.address,
+        callData: poolInterface.encodeFunctionData('feePercentage', []),
+      },
+    ])
+    .flat();
+  const poolDataCalled = await multiContract.methods
```

Comment:
As discussed in dm, we can't query the aggregated priceFeed contract through our multicall contract due to whitelisting so 2 rpc are necessary  ✅ 

I suggest a minor micro-optimisation still:
```js
const [pricesFeedResults, poolDataCalled] = await Promise.all([
  _getPricesFeedValue(
    dexHelper, 
    poolConfigs, 
    priceFeedAddress, 
    blockNumber),
  multiContract
  .methods()
  .aggregate(PoolCallData)
  .call({}, blockNumber)
]);
```